### PR TITLE
Merge DEV9ghrz into core

### PR DESCRIPTION
**Pull request by GovanifY**
As we all know, ethernet was an utter failure(see picture below) and that's the reason why we did not want to include it by default in PCSX2.
But plugin merging time has come, and so it is time to merge yet another plugin into the core!

~~I _highly_ recommend merging https://github.com/PCSX2/pcsx2/pull/3756 and waiting for me to rebase this PR before you review stylistic differences(eg tabs, brackets, etc etc) since the SPU2 PR formats core files which DEV9 modifies, but DEV9 is currently based on the master branch which does not yet have those modifications.~~
Ready for review!

image

### CODE REVIEW

File: pcsx2/DEV9/DEV9.h [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+#define SMAP_R_EMAC3_MODE1 (SMAP_EMAC3_REGBASE + 0x04)
+#define SMAP_R_EMAC3_MODE1_L (SMAP_EMAC3_REGBASE + 0x04)
+#define SMAP_R_EMAC3_MODE1_H (SMAP_EMAC3_REGBASE + 0x06)
+#define SMAP_E3_FDX_ENABLE (1 << 31)
+#define SMAP_E3_INLPBK_ENABLE (1 << 30) /* internal loop back */
+#define SMAP_E3_VLAN_ENABLE (1 << 29)
+#define SMAP_E3_FLOWCTRL_ENABLE (1 << 28) /* integrated flow ctrl(pause frame) */
+#define SMAP_E3_ALLOW_PF (1 << 27)        /* allow pause frame */
+#define SMAP_E3_ALLOW_EXTMNGIF (1 << 25)  /* allow external management IF */
+#define SMAP_E3_IGNORE_SQE (1 << 24)
```
Comment (tellowkrinkle) [inline pcsx2/DEV9/DEV9.h:257]: I think we should keep the old formatting here, the tree-like format is important

File: pcsx2/DEV9/Win32/tap-win32.cpp [hunk 2]
```diff
@@ -37,33 +26,33 @@
...
-#define TAP_IOCTL_GET_MAC               TAP_CONTROL_CODE (1, METHOD_BUFFERED)
-#define TAP_IOCTL_GET_VERSION           TAP_CONTROL_CODE (2, METHOD_BUFFERED)
-#define TAP_IOCTL_GET_MTU               TAP_CONTROL_CODE (3, METHOD_BUFFERED)
-#define TAP_IOCTL_GET_INFO              TAP_CONTROL_CODE (4, METHOD_BUFFERED)
-#define TAP_IOCTL_CONFIG_POINT_TO_POINT TAP_CONTROL_CODE (5, METHOD_BUFFERED)
-#define TAP_IOCTL_SET_MEDIA_STATUS      TAP_CONTROL_CODE (6, METHOD_BUFFERED)
-#define TAP_IOCTL_CONFIG_DHCP_MASQ      TAP_CONTROL_CODE (7, METHOD_BUFFERED)
-#define TAP_IOCTL_GET_LOG_LINE          TAP_CONTROL_CODE (8, METHOD_BUFFERED)
-#define TAP_IOCTL_CONFIG_DHCP_SET_OPT   TAP_CONTROL_CODE (9, METHOD_BUFFERED)
+#define TAP_IOCTL_GET_MAC TAP_CONTROL_CODE(1, METHOD_BUFFERED)
```
Comment (tellowkrinkle) [inline pcsx2/DEV9/Win32/tap-win32.cpp:32]: This is kind of unreadable

**Review by tellowkrinkle — COMMENTED** [review]
Marked a few candidates for clang-format overrides

File: pcsx2/DEV9/Win32/tap-win32.cpp [hunk 3]
```diff
@@ -1,31 +1,20 @@
...
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
  *
- *  You should have received a copy of the GNU General Public License
- *  along with this program (see the file COPYING included with this
- *  distribution); if not, write to the Free Software Foundation, Inc.,
- *  59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
```
Comment (lightningterror) [inline pcsx2/DEV9/Win32/tap-win32.cpp:13]: You shouldn't change this copyright header.

File: pcsx2/DEV9/Linux/net.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2010  PCSX2 Dev Team
```
Comment (lightningterror) [inline pcsx2/DEV9/Linux/net.cpp:2]: 2010? Same for the others.

File: pcsx2/DEV9/Win32/tap-win32.cpp (hunk 3 above)
Comment (GovanifY) [reply pcsx2/DEV9/Win32/tap-win32.cpp:13]: I rewrote a good bunch of this file, I believe this falls under fair use

Comment (lightningterror) [reply pcsx2/DEV9/Win32/tap-win32.cpp:13]: Need to discuss it with the others.

**Comment by lightningterror** [comment]
You can squash the last commit in the older formatting one, also rename the char conversion one with a prefix.

**Review by lightningterror — APPROVED** [review]
Just need someone to test it. Maybe @prafullpcsx2

**Comment by GovanifY** [comment]
@lightningterror it has already been tested by our testers before i went on a break, I fixed the only issue that was found(inconsistency between wide chars). Want me to ping back the testers again?

**Comment by GovanifY** [comment]
@lightningterror Everything ready for merge, @prafullpcsx2 confirmed the changes!

**Review by RedDevilus — APPROVED** [review]

**Pull request merged**